lib/client.js

```diff
@@ -402,7 +402,17 @@ class Client extends EventEmitter {
             });
           } else if (curAuth.type === 'publickey') {
             proto.authPK(curAuth.username, curAuth.key, (buf, cb) => {
-              const signature = curAuth.key.sign(buf);
+              let signatureAlgo;
+              if (curAuth.key.type === 'ssh-rsa') {
+                if (this._protocol._remoteHostKeyAlgorithms
+                    .includes('rsa-sha2-512')) {
+                  signatureAlgo = 'sha512';
+                } else if (this._protocol._remoteHostKeyAlgorithms
+                    .includes('rsa-sha2-256')) {
+                  signatureAlgo = 'sha256';
+                }
+              }
+              const signature = curAuth.key.sign(buf, signatureAlgo);
```

Comment:
Need to add this logic for `agent` type too

Reply:
Could you point me to the right location to add that? I don't have that much experience with the agent protocol

Reply:
It's just above this code, line 391, you can check this [commit](https://github.com/jeanp413/ssh2/commit/0c3652b7235be765579c032a9372c1ae49f2ceb3) on my fork as I needed to cherry-pick your PR :+1:

Reply:
Done